### DISCUSSION TITLE
Understanding why Glucose Direct Can't Interact Directly with Sensor

### DISCUSSION THREAD

**Discussion by riverview999** [creepymonster/GlucoseDirect#602] [General]
Thanks so much for everyone responsible in creating this app.  Very well done!

I'm a software product manager / diabetic at a major vendor (not health care related) and am struggling to understand why Abbott hasn't released a native Apple Watch app or made it possible for third parties to do so.  My opinion is Dexcom will start to take CGM market share since its watch app interacts directly with the sensor (no phone app intermediary required).  So, it makes no sense to me Abbott is not publicly disclosing a roadmap which shows native watch app implementation in their roadmap.

In your opinion, what is preventing you or Abbott itself from creating a native app?  Are there API issues, concerns about exposing raw data, issues around maintaining current Bluetooth session, something related to the Libre 3 sensor itself, another issue?

If I can understand what exactly is going on I'm hopeful I can use industry contacts to help Abbott make the right decision around facilitating a native watch app creation by either third parties or itself.

**Comment by DAcodedBEAT** [comment]
@riverview999 I'm not the maintainer of this project but I can answer a few questions as a fellow onlooker.  Abbott has not opened up their CGM and from previous accounts, have been [very aggressive against attempts at reverse engineering made public](https://github.com/NightscoutFoundation/xDrip/issues/1474#issuecomment-1041327847). I agree that it doesn't make sense but that is how they decided to operate.

Decrypting the BLE data from the CGM (which Abbott safeguards) and fear of a DMCA takedown from Abbott prevents the open source community from creating a native app, though there is nothing preventing Abbott from either opening this up or making improvements to their app.

**Comment by gui-dos** [comment]


**Reply by philipp-1337** [reply]
@gui-dos thanks for sharing the link to your watchOS App. Unfortunately it requires iOS 18 which I'm not willing to update to just yet. Is there any other way?
